articles/templating-your-static-site/index.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: templating-your-static-site
+title: Templating Your Static Site (Converting a Static Site to a Static Site Generator)
+description: Tutorial on converting a simple static website to a static site generator, Eleventy and using EJS templating.
+author: louise-findlay
+date: 2020-07-15T00:00:00-07:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/templating-your-static-site/hero.jpg
+    alt: templating static sites eleventy
+---
+Static sites are much easier to develop and deploy than full-stack dynamic web apps. You don’t have to pay monthly hosting charges (if you use a free hosting provider like [Netlify](https://www.netlify.co) and if you use a CSS framework like [Bootstrap](https://getbootstrap.com) or [Materialize](https://materializecss.com) it only takes a basic knowledge of HTML and CSS to code a simple website. **Insert reference to articles you’ll write about designing and developing static websites**
+
+<!--more-->
+
+However, for more complex multi-page websites, a static site may not be suitable. Every time you want to add content, a new page has to be created and coded. If you change something that appears on multiple pages then you’d have to change it everywhere. That’s where static site generators come in. They bring the power of templating to static sites.
+
+There are many static site generators, [Hugo](https://gohugo.io), [Gatsby](http://gatsbyjs.org) and [Next.js](http://gatsbyjs.org) to name a few but we will use Eleventy because it’s easy to use and flexible. It uses vanilla JavaScript and allows multiple template languages to be mixed and matched so you don’t have to learn a new language or templating system. Static site generators work by taking the layout templates you’ve created and building the pages using the template and data you’ve provided.
```

Comment:
Your link for next.js is incorrect. 

Also, I would not consider this a requirement and is definitely nit-picky, but your link to Gatsby is not for the secure site (https). I have yet to see an official site that doesn't automatically redirect to https and often they also redirect to the www subdomain as well. As you move forward in this tech space consider links that will not automatically be redirected a couple of times (meaning go to the site, let it load, and then grab the final URL once it's fully loaded - this will be more important if you end up working with web performance and testing).

Reply:
Thanks for the advice/suggestions. I've corrected those URLs and added one in for Eleventy.